File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public const int CARS_PER_HOUR = 221;
    
    public static double SuccessRate(int speed)
    {
        if (speed == 0) {
            return 0;
        }
        else if (speed <= 4) {
            return 1;
        }
        else if (speed <= 8) {
            return 0.9;
        }
        else if (speed == 9) {
            return 0.8;
        }
        else {
            return 0.77;
        }
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return SuccessRate(speed) * speed * CARS_PER_HOUR;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return Convert.ToInt32(ProductionRatePerHour(speed)) / 60;
    }
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public static int CARS_PER_HOUR = 221;
    
    public double productionRatePerHour(int speed) {
        if (speed <= 4) {
            return CARS_PER_HOUR * speed;
        }
        else if (speed <= 8) {
            return CARS_PER_HOUR * speed * 90.0 / 100;
        }
        else if (speed == 9) {
            return CARS_PER_HOUR * speed * 80.0 / 100;
        }

        return CARS_PER_HOUR * speed * 77.0 / 100;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}
